src/dev/src/app/components/tree-view/tree-view-client-side-demo.ts

```diff
@@ -0,0 +1,480 @@
+  NonRootTreeNode,
+} from '@ferui/components';
+
+@Component({
+  template: `
+    <h1 class="mb-4">FerUI Tree View</h1>
+    <h3>Overview</h3>
+    <p>
+      Just like any other FerUI component in <code>@ferui-components</code>, the Tree View will require adding the FerUI to
+      your project's main module to use this component like a regular Angular component.
+    </p>
+    <p>Import the FeruiModule in your app.module file and in your imports array module</p>
+    <p>In order to use the Tree View Component you will need:</p>
+    <ul>
+      <li>Data object which implements a <code>TreeNodeData interface</code></li>
+      <li>Data retriever object which implements a <code>TreeNodeDataRetriever interface</code></li>
+      <li>Configuration object which implements a <code>TreeViewConfiguration interface</code></li>
+    </ul>
+    <p>
+      Ferui Tree View Component allows developer to subscribe to an event observable that triggers when a Tree Node is
+      expanded, collapsed, or selected. Expects to receive a TreeViewEvent.
+    </p>
+    <p>Three examples follow</p>
+
+    <div class="demo-tree-view">
+      <h1>Client Side Tree View</h1>
+      <div class="demo-component">
+        <fui-tree-view
+          [treeNodeData]="treeNodeData"
+          [dataRetriever]="treeDataRetriever"
+          [config]="{ width: '250px', height: '300px' }"
+        ></fui-tree-view>
+      </div>
+      <pre class="code-example"><code [languages]="['typescript']" [highlight]="dataExample1"></code></pre>
+    </div>
+    <div class="demo-tree-view">
+      <h1>Server Side Tree View With Borders</h1>
+      <div class="demo-component">
+        <fui-tree-view
+          [loading]="loading"
+          [treeNodeData]="serverSideTreeNodeData"
+          [dataRetriever]="serverDataRetriever"
+          [config]="{ width: '250px', height: '300px', hasBorders: true }"
+        ></fui-tree-view>
+      </div>
+      <pre class="code-example"><code [languages]="['typescript']" [highlight]="dataExample2"></code></pre>
+    </div>
+    <div class="demo-tree-view">
+      <h1>Client Side Tree View Non Root</h1>
+      <div class="demo-component">
+        <fui-tree-view
+          [treeNodeData]="noRoot"
+          [dataRetriever]="treeDataArrayRetriever"
+          [config]="{ width: '250px', height: '300px' }"
+        ></fui-tree-view>
+      </div>
+      <pre class="code-example"><code [languages]="['typescript']" [highlight]="dataExample3"></code></pre>
+    </div>
+
+    <ng-template #expandedFolder>
+      <clr-icon class="fui-less-icon" shape="fui-solid-arrow"></clr-icon>
+    </ng-template>
+    <ng-template #nonExpandedFolder>
+      <clr-icon class="fui-add-icon" shape="fui-solid-arrow"></clr-icon>
+    </ng-template>
+
+    <ng-template #template let-node="node">
+      <span>
+        {{ node.data.data[node.data.label] }}
+        <clr-icon class="fui-add-icon" shape="fui-screenshot"></clr-icon>
+      </span>
+    </ng-template>
+  `,
```

Comment:
Can you also add 2 more examples with an already opened (expanded) server-side and client-side treeView please ? 
This will be needed.